notebooks/standardize_events_data.ipynb

```diff
@@ -90,90 +297,914 @@
+    "                'token': data_list[0],\n",
+    "                'lending_accumulator': data_list[1],\n",
+    "                'debt_accumulator': data_list[2]\n",
+    "            })\n",
+    "        else:\n",
+    "            return pd.Series()\n",
+    "    except Exception as e:\n",
+    "        # Handle any parsing errors\n",
+    "        return pd.Series()\n",
+    "\n",
+    "# Filter the dataset for 'zklend' protocol\n",
+    "zk_data = df_events[df_events['Protocol'] == 'zklend']\n",
+    "\n",
+    "# Apply the extraction function to each row\n",
+    "extracted_elements = zk_data.apply(extract_zklend, axis=1)\n",
+    "\n",
+    "# Concatenate the extracted elements with the original dataframe\n",
+    "standard_zk = pd.concat([zk_data[['block_hash', 'block_number', 'transaction_hash', 'timestamp', 'key_name']], extracted_elements], axis=1)\n",
+    "\n",
+    "# display the dataframe\n",
+    "# standard_zk.head()\n",
+    "\n",
+    "# Uncomment the following below to Save the resulting dataframe to a new CSV file\n",
+    "# output_file_path = 'extracted_standard_zk_events.csv'\n",
+    "# standard_zk.to_csv(output_file_path, index=False)"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "0737a215-bd96-49d9-b236-215060d00558",
+   "id": "99932d49-9099-4bed-b81f-e733abf42269",
    "metadata": {},
    "source": [
-    "4. Apply the Transformation Function to the DataFrame\n",
-    "Apply the decode_and_convert_row function to each row of the DataFrame to create a new standardized DataFrame."
+    "### For nostra_alpha protocol, these are the events and their relevant data\n",
+    "- for debt event: user is in data[0], token is in from_address, face_amount is in data[1]\n",
+    "- for mint event: user is in data[0], token is in from_address, face_amount is in [1]\n",
+    "- for interest state updated event :  debtToken is in data[0], lendingRate is in data[1], borrowRate is in data[3], lendIndex is in [5], borrowIndex is in [7]"
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": null,
-   "id": "6dc5df33-6f26-4cad-a4a6-9d3c42e64559",
+   "execution_count": 7,
+   "id": "11dc202e-2df0-48e1-9d02-b4384c0412d8",
    "metadata": {},
    "outputs": [],
    "source": [
-    "# Apply the function to each row\n",
-    "standardized_rows = df.apply(decode_and_convert_row, axis=1)\n",
-    "standardized_df = pd.DataFrame(standardized_rows.tolist())"
+    "# Define a function to decode bytes, convert to list, and extract specific elements based on the event\n",
+    "def extract_nostra_alpha(row):\n",
+    "    \n",
+    "    def hex_to_int(hex_str):\n",
+    "        return int(hex_str, 16)\n",
+    "    \n",
+    "    try:\n",
+    "        # Decode the bytes to a string and convert the string to an actual list\n",
+    "        data_str = row['data'].decode('utf-8')\n",
+    "        data_list = eval(data_str)\n",
+    "   \n",
+    "        if row['key_name'] == 'Mint':\n",
```

Comment:
From this, we don't know whether the `Mint` event represents a deposit or borrowing event. Pls add that information by mapping `token` on the type of token defined here: https://github.com/CarmineOptions/derisk-research/blob/master/src/nostra_alpha.py#L36.

For example, `Mint` events for `non_interest_bearing_collateral` or `interest_bearing_collateral` are deposit events, for `debt` they're borrowing events, etc. Map the event type names on those used for zkLend.